src/dlfcn.c

```diff
@@ -575,6 +575,11 @@ static void *get_address_from_import_address_table( void *iat, DWORD iat_size, v
     if( !is_valid_address( ptr ) ||  ptr < (BYTE *)iat || ptr > (BYTE *)iat + iat_size )
         return NULL;
 
+    /* limit checks to mingw for now as msvc seems to have a different binary layout */
+#ifdef __MINGW32__
+    if( ptr < (BYTE *)iat || ptr > (BYTE *)iat + iat_size )
+        return NULL;
+#endif

```

Comment:
This check would not work as it is compile time check. You would rather need runtime check if caller of dladdr() was compiled under mingw or not. And this is hard to check, probably impossible. Because you can compile libdl.dll library under compiler A, project exeutable file by compiler B and library function stored in mylib.dll which address is put into dladdr() by compiler C.